frontend/src/AnalyticsPage.tsx

```diff
@@ -0,0 +1,120 @@
+                <PostAnalyticsTile analyticsNumber={post} label="# Shares" eventBus={this.props.eventBus} keyVal="shares" type={AnalyticsType.POST} background="share" />
+                <PostAnalyticsTile analyticsNumber={post} label="# Followers" eventBus={this.props.eventBus} keyVal="acc_followers" type={AnalyticsType.ACCOUNT} background="follower" />
+                <PostAnalyticsTile analyticsNumber={post} label="# Posts" eventBus={this.props.eventBus} keyVal="acc_posts" type={AnalyticsType.ACCOUNT} background="post" />
+            </div>
+        );
+    }
+
+    /***Called by the Multiselect UI element. Synchronizes the multiselect state to the component state. */
+    private onSelectionChange(value: MultiSelectValue[]) {
+        let accounts = value.filter(v => v.type === "Accounts").map(v => this.state.accounts.find(sa => v.value === sa.id));
+        let posts = [];//value.filter(v => v.type === "Posts").map(v => this.state.posts.find(sa => v.value == sa.id));
+        this.updateData(EventBusEventType.REFRESH_POST_ANALYTICS_REQ, "", accounts as AccountDataSource[], posts as PostDataSource[]);
+        this.updateData(EventBusEventType.REFRESH_ACCOUNT_ANALYTICS_REQ, "", accounts as AccountDataSource[]);
+    }
+
+    /** Called by the event handler when new data are to be fetched. */
+    private updateData(eventType: EventBusEventType, relId: string, selectedAccounts?: AccountDataSource[], selectedPosts?: PostDataSource[]) {
+        let typedPath = eventType === EventBusEventType.REFRESH_ACCOUNT_ANALYTICS_REQ ? "account" : "post";
+        let params = (selectedPosts ? selectedPosts.map(p => p ? "&posts=" + p.id : "").join("") : "") + (selectedAccounts ? selectedAccounts.map(a => a ? "&accounts=" + a.id : "").join("") : "");
+        fetch(DeploymentManager.getUrl() + "api/analytics/" + typedPath + "/" + relId + "?query=" + params, {
+            method: 'get',
+            credentials: 'include',
+        })
+            .then(response => {
+                if (!response.ok) {
+                    ToastManager.showErrorToast(response);
+                } else {
+                    response.json().then(data => {
+                        if (eventType === EventBusEventType.REFRESH_ACCOUNT_ANALYTICS_REQ) {
+                            this.props.eventBus.fireEvent(EventBusEventType.REFRESH_ACCOUNT_ANALYTICS, data);
+                            this.setState({ accounts: data[0].entities.concat(data[0].filtered), selectedAccounts: data[0].filtered.length ? data[0].entities : [] });
+                        }
+                        else {
+                            this.props.eventBus.fireEvent(EventBusEventType.REFRESH_POST_ANALYTICS, data);
+                            this.setState({ posts: data[0].entities.concat(data[0].filtered), selectedPosts: data[0].filtered.length ? data[0].entities : [] });
+                        };
+                    });
+                }
+            });
+    }
```

Comment:
You could avoid nesting and indenting code too much by using [`async`/`await`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/async_function#Description) 
```suggestion        
        let response = await fetch(DeploymentManager.getUrl() + "api/analytics/" + typedPath + "/" + relId + "?query=" + params, {
            method: 'get',
            credentials: 'include',
        })
        if (!response.ok) {
            ToastManager.showErrorToast(response);
        } else {
            let data = await response.json();
                if (eventType === EventBusEventType.REFRESH_ACCOUNT_ANALYTICS_REQ) {
                    this.props.eventBus.fireEvent(EventBusEventType.REFRESH_ACCOUNT_ANALYTICS, data);
                    this.setState({ accounts: data[0].entities.concat(data[0].filtered), selectedAccounts: data[0].filtered.length ? data[0].entities : [] });
                }
                else {
                    this.props.eventBus.fireEvent(EventBusEventType.REFRESH_POST_ANALYTICS, data);
                    this.setState({ posts: data[0].entities.concat(data[0].filtered), selectedPosts: data[0].filtered.length ? data[0].entities : [] });
                };
            });
        }
```